core/app/components/LastVisitedCommunity/SetLastVisited.tsx

```diff
@@ -0,0 +1,16 @@
+"use client";
+
+import { useEffect } from "react";
+
+import { setLastVisited } from "~/app/components/LastVisitedCommunity/lastVisited";
+import { useServerAction } from "~/lib/serverActions";
+
+export default function SetLastVisited({ communitySlug }: { communitySlug: string }) {
+	const runSetLastVisited = useServerAction(setLastVisited);
+
+	useEffect(() => {
+		runSetLastVisited(communitySlug);
+	}, [communitySlug, runSetLastVisited]);
+
+	return <></>;
+}
```

Comment:
I think doing this on the client is probably easier, since we don't need it during the initial render on the server (ie we dont need the cookie info after it's set). it's the same way the (new) sidebar does it to remember open state.

```suggestion
import { LAST_VISITED_COOKIE, LAST_VISITED_COOKIE_MAX_AGE } from "./constants";

export default function SetLastVisited({ communitySlug }: { communitySlug: string }) {
	useEffect(() => {
		document.cookie = `${LAST_VISITED_COOKIE}=${communitySlug}; path=/; max-age=${LAST_VISITED_COOKIE_MAX_AGE}`;
	}, [communitySlug]);

	return <></>;
}
```

Reply:
there is possibly some more elegant way of doing this, but this does seem to work.

Reply:
based on this code from the new sidebar

https://github.com/pubpub/platform/blob/8f04e3a3fec92e0c18611199874bdad02c7ed7d4/packages/ui/src/sidebar.tsx#L77-L91